.circleci/config.yml

```diff
@@ -53,9 +54,9 @@ workflows:
         matrix:
           parameters:
             consul_version:
-            - "1.16"
-            - "1.17"
-            - "1.18"
+            - "1.16.6"
```

Comment:
I don't think we want to maintain which patch release is specified in CI. This is a bit more toil than we want to maintain.

Is there a good reason to do it this way?

What about a CI step that gets the version of consul from the binary?

It looks like the Consul dockerfile includes an environment variable `PRODUCT_VERSION` which could be used.

We could also do something like this:
```
echo "CONSUL_VERSION=$(/bin/consul version | awk '$1 == "Consul" {print $2}' | tr -d 'v') >> $BASH_ENV
```
This way we can keep the minor version and detect the patch version on-demand.

See: https://circleci.com/docs/set-environment-variable/#set-an-environment-variable-in-a-shell-command

Reply:
cmiiw, this mean we don't need to configure the CONSUL_VERSION env in the circle CI config and just use PRODUCT_VERSION directly on the test?

Reply:
updated using $BASH_ENV, feel free to check @SuperQ

Reply:
I think keeping the `CONSUL_VERSION` var makes it a little more clear to people testing outside Docker.